README.md

```diff
@@ -101,34 +100,46 @@ Returns a `prometheus` object that should be used to register metrics.
 
 Example:
 ```
-init_by_lua '
+init_worker_by_lua_block {
   prometheus = require("prometheus").init("prometheus_metrics")
-';
+}
 ```
 
 ### init_worker()
 
 **syntax:** prometheus:init_worker([*sync_interval*])
 
-Initializes per-worker counter. This must be called from the
-`init_worker_by_lua` section of nginx configuration.
+Initializes per-worker counter. This is typically done automatically by the
+`init` function if it's called from `init_worker_by_lua_block`. However,
+if you would like to override `sync_interval`, you can call the main `init`
+function in the [init_by_lua_block](
+https://github.com/openresty/lua-nginx-module#init_by_lua_block) section, and
+then manually call `prometheus:init_worker` in
+[init_worker_by_lua_block](https://github.com/openresty/lua-nginx-module#init_worker_by_lua_block).
```

Comment:
This seems quite over-complicated. It would make more sense if the sync_interval could be specified directly in the `init` function. I understand that you do this to keep it backward compatible. But I believe it should be possible to make it configurable from `init`, so that calling `init_worker` would become deprecated (and eventually could be removed from code, or at least from documentation).

Since there is now multiple configuration options, I'd suggest to allow passing them as a table:

```lua
function Prometheus.init(dict_name, options, error_metric_name)
  -- ...
  if type(options) == "table" then
    self.prefix = options.prefix or ''
    self.error_metric_name = options.error_metric_name or DEFAULT_ERROR_METRIC_NAME
    self.sync_interval = options.sync_interval or 1
  else
    self.prefix = options
    self.error_metric_name = error_metric_name or DEFAULT_ERROR_METRIC_NAME
  end
  -- ...
```
This would make it more extensible to the future and also the calls would be more user-friendly, since you could omit the values where default is used.

Reply:
I like the idea of accepting a table of options. I am tempted to remove the third option to `init()`, since it was [added just a few days ago](https://github.com/knyar/nginx-lua-prometheus/pull/91) (and we've not had a tagged release since) and make it passed via options as well.

Reply:
I agree, go ahead. Since the code was not officially released yet, I'm possibly the only one actually using it. And I didn't even release the packages we use inside my company, so feel free to remove the `error_metric_name`.

It would be nice if you'd tag a release after this is done.

Reply:
Cool, I've done this. I've also made the `sync_interval` log a warning if it's called twice as you suggested. PTAL?

I'd like to wait a few days after merging this, and then I will tag a new release if that's OK with you.

Reply:
Yes, I did look and I'm happy :slightly_smiling_face:

The release doesn't date doesn't matter much to me. I use this repo as submodule, so I can easily point it to any commit even if it is not tagged as release yet.